commit 12ab16597189993deb6a17827e137c0357ef0de6
Author: Michael Zoran <mzoran@crowfest.net>
Date:   Tue Mar 7 19:23:36 2017 -0800

    staging: vchiq_arm: Disable ability to dump memory by default
    
    vc04_services has an ioctl interface to dump arbitrary memory
    to a custom debug log.  This is typically only needed by
    diagnostic tools, and can potentially be a security issue
    if the devtmpfs node doesn't have adequate permissions set.
    
    Since the ability to dump memory still has debugging value,
    create a new build configuration and disable the feature
    by default.
    
    Signed-off-by: Michael Zoran <mzoran@crowfest.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/staging/vc04_services/Kconfig b/drivers/staging/vc04_services/Kconfig
index eb52cec18c87..b9f316603308 100644
--- a/drivers/staging/vc04_services/Kconfig
+++ b/drivers/staging/vc04_services/Kconfig
@@ -18,6 +18,18 @@ config BCM2835_VCHIQ
 		Defaults to Y when the Broadcom Videocore services
 		are included in the build, N otherwise.
 
+if BCM2835_VCHIQ
+
+config BCM2835_VCHIQ_SUPPORT_MEMDUMP
+	bool "Support dumping memory contents to debug log"
+	help
+		BCM2835 VCHIQ supports the ability to dump the
+		contents of memory to the debug log.  This
+		is typically only needed by diagnostic tools used
+		to debug issues with VideoCore.
+
+endif
+
 source "drivers/staging/vc04_services/bcm2835-audio/Kconfig"
 
 source "drivers/staging/vc04_services/bcm2835-camera/Kconfig"